darshan-runtime/lib/darshan-core.c

```diff
@@ -75,6 +76,11 @@ static int mnt_data_count = 0;
 extern void bgq_runtime_initialize();
 #endif
 
+#ifdef HAVE_LDMS
+extern struct darshanConnector dC;
+extern void darshan_ldms_connector_initialize();
+#endif
```

Comment:
If we move all of the LDMS code into the initialize routine as I suggest below, we can remove this block. No need to define external variable `dC` in that case. Also, there shouldn't be any need to declare the initialize routine as an extern in any case, since it's defined as a prototype in the header included above.